pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -653,8 +654,9 @@ impl RemoteTimelineClient {
         // to syntactically forbid ? or bail! calls here.
         let no_bail_here = || {
             for name in names {
-                upload_queue.latest_files.remove(name);
-                upload_queue.latest_files_changes_since_metadata_upload_scheduled += 1;
+                if upload_queue.latest_files.remove(name).is_some() {
+                    upload_queue.latest_files_changes_since_metadata_upload_scheduled += 1;
+                }
```

Comment:
This I think is a semi-important change, could split off.

I don't really depend on it.

Reply:
Split off to #5103.